prow/config/jobs_test.go

```diff
@@ -168,38 +256,39 @@ func TestCommentBodyMatches(t *testing.T) {
 			"org/repo": {
 				{
 					Name:      "gce",
-					re:        regexp.MustCompile(`@k8s-bot (gce )?test this`),
+					re:        regexp.MustCompile(`(?m)^/test( all| gce),?(\s+|$)`),
 					AlwaysRun: true,
 				},
 				{
 					Name:      "unit",
-					re:        regexp.MustCompile(`@k8s-bot (unit )?test this`),
+					re:        regexp.MustCompile(`(?m)^/test( all| unit),?(\s+|$)`),
 					AlwaysRun: true,
 				},
 				{
 					Name:      "gke",
-					re:        regexp.MustCompile(`@k8s-bot (gke )?test this`),
+					re:        regexp.MustCompile(`(?m)^/test( all| gke),?(\s+|$)`),
 					AlwaysRun: false,
 				},
 				{
 					Name:      "federation",
-					re:        regexp.MustCompile(`@k8s-bot federation test this`),
+					re:        regexp.MustCompile(`(?m)^/test federation,?(\s+|$)`),
 					AlwaysRun: false,
 				},
 			},
 			"org/repo2": {
 				{
 					Name:      "cadveapster",
-					re:        regexp.MustCompile(`@k8s-bot test this`),
+					re:        regexp.MustCompile(`(?m)^/test all,?(\s+|$)`),
 					AlwaysRun: true,
 				},
 			},
 		},
 	}
 	for _, tc := range testcases {
-		actualJobs := c.MatchingPresubmits(tc.repo, tc.body, regexp.MustCompile(`ok to test`))
+		actualJobs := c.MatchingPresubmits(tc.repo, tc.body, regexp.MustCompile(`(?m)^/ok-to-test,?(\s+|$)`))
```

Comment:
But this doesn't have anything to do with validating `config.yaml`, since you're passing in the ok to test regex definition into c.MatchingPresubmits() here. It just validates the behavior of the MatchingPresubmits with fake data.